.pre-commit-config.yaml

```diff
@@ -0,0 +1,35 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+default_language_version:
+    python: python3.7
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v2.4.0
+    hooks:
+    -   id: no-commit-to-branch
```

Comment:
I'm not sure about this one. There are sometimes you may commit to master, although it's not frequent I suppose.

Reply:
You can always skip a check.Docs should still be updated with pointers to pre-commit.com documentation (because you can also say to skip single check instead of the whole hook).

Reply:
Probably both. I'm more likely to remember to use `-n` than the specific syntax to disable a specific hook and having to remember its exact name.

Reply:
But I guess most of the time commiting to master is not desired so I guess it's fine as long as we doc how to turn off the hook.

Reply:
the hook name is shown when it fails so remembering it is not a problem :) Also committing/pushing directly to master should be so extraordinary as not to  matter the slightest if it's difficult, way bigger problems have come up if direct pushing to master is *actually needed* to solve an issue.